crates/marine-build-rs-generator/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#![doc(html_root_url = "https://docs.rs/marine-build-rs-generator/0.4.0")]
+#![deny(
+    dead_code,
+    nonstandard_style,
+    unused_imports,
+    unused_mut,
+    unused_variables,
+    unused_unsafe,
+    unreachable_patterns
+)]
+#![warn(rust_2018_idioms)]
+#![recursion_limit = "1024"]
+
+use marine_test_macro_impl::generate_marine_test_env_impl;
+use std::collections::HashMap;
+
+pub use marine_test_macro_impl::ServiceDescription;
+use std::path::{PathBuf, Path};
+use std::{fs, env};
+
+pub fn generate_marine_test_env(
+    services: &[(String, ServiceDescription)],
+    filename: &str,
+    build_rs_file_path: &str,
+) {
+    let services = services
```

Comment:
Probably you won't needed this after changing `generate_service_definitions` to recieve iterator instead of `HashMap`.